File: 1-reverse-number/appositum/reverse_number.py
from functools import reduce

def reverse(num):
    """Return a reversed integer.

    >>> reverse(1234)
    4321
    >>> reverse(12)
    21
    >>> reverse(0)
    0
    >>> reverse(6641112222223333333345678901118)
    8111098765433333333222222111466
    >>> reverse('lalala')
    'lalala is not a number!'
    """

    if type(num) != int:
        return num + ' is not a number!'

    lst = reduce(lambda acc, n: [n] + acc, str(num), [])
    return int(''.join(lst))


if __name__ == '__main__':
    import doctest
    doctest.testmod()

File: 1-reverse-number/caioluiz/reverse-number.exs
defmodule Solution do
  def reverse_number(num) when is_integer(num) do
    num
    |> Integer.to_charlist
    |> Enum.reduce([], fn x, acc -> [x | acc] end)
    |> List.to_integer
  end
  
  def reverse_number(something_else), do: "#{something_else} is not a number"
end

IO.puts Solution.reverse_number 123
IO.puts Solution.reverse_number 12
IO.puts Solution.reverse_number 0
IO.puts Solution.reverse_number 6641112222223333333345678901118
IO.puts Solution.reverse_number 'lalala'
